library/Freckle/App/Memcached/Client.hs

```diff
@@ -48,30 +51,49 @@ memcachedClientDisabled :: MemcachedClient
 memcachedClientDisabled = MemcachedClientDisabled
 
 get
-  :: (MonadIO m, MonadReader env m, HasMemcachedClient env)
+  :: (MonadUnliftIO m, MonadTracer m, MonadReader env m, HasMemcachedClient env)
   => CacheKey
   -> m (Maybe Value)
-get k = with $ \case
+get k = traced $ with $ \case
   MemcachedClient mc -> liftIO $ view _1 <$$> Memcache.get mc (fromCacheKey k)
   MemcachedClientDisabled -> pure Nothing
+ where
+  traced =
+    inSpan
+      "cache.get"
+      clientSpanArguments
+        { Trace.attributes = HashMap.fromList [("key", Trace.toAttribute k)]
+        }
 
 -- | Set a value to expire in the given seconds
 --
 -- Pass @0@ to set a value that never expires.
 set
-  :: (MonadIO m, MonadReader env m, HasMemcachedClient env)
+  :: (MonadUnliftIO m, MonadTracer m, MonadReader env m, HasMemcachedClient env)
   => CacheKey
   -> Value
   -> CacheTTL
   -> m ()
-set k v expiration = with $ \case
+set k v expiration = traced $ with $ \case
   MemcachedClient mc ->
     void $
       liftIO $
         Memcache.set mc (fromCacheKey k) v 0 $
           fromCacheTTL
             expiration
   MemcachedClientDisabled -> pure ()
+ where
+  traced =
+    inSpan
+      "cache.set"
+      clientSpanArguments
+        { Trace.attributes =
+            HashMap.fromList
```

Comment:
Offtopic/unimportant, but any interest in adding `GHC.Exts.fromList` to the prelude? Could have spared you an import here.

Reply:
I'm into it. I've been trying to migrate `SomeModule.toList` to the (already prelude-available) one from `Data.Foldable` as a similar simplification.